### PR TITLE
Optimagic and other modernisation

### PR DESCRIPTION
**Pull request by hmgaudecker**
- [x] use optimagic instead of estimagic, get rid of deprecations
- [x] get rid of Pandas 2.2 deprecations
- [x] modernise infrastructure (pixi etc.)
- [x] provision tests for profiling memory consumption

### CODE REVIEW

**Comment by hmgaudecker** [comment]
I think it would be good to make perfomance checks on this basis -- I have not changed functionality or anything (other than deleting Jacobian entries from `maximization_inputs`), mainly the way `get_maximization_inputs` works internally and making things future-proof.

**Comment by hmgaudecker** [comment]
I don't think I have permissions to change the checks, those would need updating.

**Comment by janosg** [comment]
I moved the repo to OSE where you have admin permissions.

File: README.md [hunk 1]
```diff
@@ -1,65 +1,61 @@
...
+
+- clone the repository and make your changes
+- Install conda-build 2.0 or higher (earlier versions won't work!)
+- Adjust the version number in setup.py and meta.yaml.
+- Open a shell in the directory that contains skillmodels (i.e. one level higher than
+  your git repository)
+- type: conda build skillmodels
+
+For more information see the
+[conda documentation](http://conda.pydata.org/docs/building/build.html)
```
Comment (janosg) [inline README.md]: This is outdated and can be removed; But we should add a section on pixi commands instead.

**Review by janosg — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (hmgaudecker) [reply README.md]: Thanks, will do as part of addressing #74 !

**Pull request merged**